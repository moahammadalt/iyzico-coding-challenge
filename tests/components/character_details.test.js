import React, { Component } from 'react';
import { BrowserRouter as Router, Link } from "react-router-dom";
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CharacterDetails from '../../src/components/character_details';

const mockStore = configureStore();
  
const initialState = {
  selectedCharacter: {}
};

let selectedCharacter = {
  id: '1',
  name: 'test',
  origin: {
    name: 'test'
  },
  episodesList: [
    {name: 'test1'},
    {name: 'test2'}
  ]
}

const setInitialState = state => state ? {...initialState, ...state} : {...initialState};

// Snapshots
describe('<CharacterDetails /> Snapshot test for CharacterDetails component', () => {

  it('Should match the CharacterDetails snapshot when there is no character data', () => {

    const store = mockStore(setInitialState());

    const wrapper = shallow(
      <Provider store={store}>
        <CharacterDetails />
      </Provider>
    );

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should match the CharacterDetails snapshot when there is character data', () => {

    const store = mockStore(
      setInitialState({ selectedCharacter })
    );

    const wrapper = shallow(
      <Provider store={store}>
        <CharacterDetails />
      </Provider>
    );

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });

});