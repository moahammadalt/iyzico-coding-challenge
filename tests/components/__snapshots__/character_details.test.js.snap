// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterDetails /> Snapshot test for CharacterDetails component Should match the CharacterDetails snapshot when there is character data 1`] = `
<div
  className="col-md-11"
>
  <div
    className="col-md-6"
  >
    <img
      alt="sorry, not available now"
      className="character-img-left rounded"
    />
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="character-details-right"
    >
      <div
        className="character-detail-field border-bottom row"
      >
        <label
          className="col-sm-4"
        >
          NAME:
        </label>
        <p
          className="col-sm-8"
        >
          test
        </p>
      </div>
      <div
        className="character-detail-field border-bottom row"
      >
        <label
          className="col-sm-4"
        >
          ORIGIN:
        </label>
        <p
          className="col-sm-8"
        >
          test
        </p>
      </div>
      <div
        className="character-detail-field border-bottom row"
      >
        <label
          className="col-sm-4"
        >
          EPISODES:
        </label>
        <div
          className="col-sm-8"
        >
          <p>
            test1
          </p>
          <p>
            test2
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CharacterDetails /> Snapshot test for CharacterDetails component Should match the CharacterDetails snapshot when there is no character data 1`] = `
<div
  className="col-md-11"
>
  <div
    className="col-md-6"
  >
    <img
      alt="sorry, not available now"
      className="character-img-left rounded"
    />
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="character-details-right"
    />
  </div>
</div>
`;
